src/odemis/driver/simcam.py

```diff
@@ -289,19 +329,69 @@ def _simulate(self):
         binning = self.binning.value
         res = self.resolution.value
         pxs_pos = self.translation.value
-        shape = self._img.shape
+
+        if self._stage:
+            pos = self._stage.position.value["x"], self._stage.position.value["y"]
+
+            exshift = numpy.array(self._orig_stage) - numpy.array(pos)
+            exshift = tuple(exshift)
+
+            # #convert to pxl
+            pxs = self._imgs[0].metadata[model.MD_PIXEL_SIZE]
+            X = exshift[0]/pxs[0]
+
+            Y = exshift[1] / pxs[1]
+            exs = (X,Y)
+
+            npos=numpy.array(pxs_pos) + numpy.array(exs)
+            pxs_pos=tuple(npos)
+
+            logging.debug('the position is %s',pxs_pos)
+
+        # Pick the image based on the current excitation wl
+
+        try:
+            current_inwl = self.getMetadata()[model.MD_IN_WL]   # (350e-9, 360e-9)
+            logging.debug('current_inwl %s',current_inwl)
```

Comment:
Better move this log message after the selection, indicating the current wl and the wavelength picked.